lib/srv/alpnproxy/local_proxy_test.go

```diff
@@ -101,6 +104,34 @@ func TestHandleAWSAccessSigVerification(t *testing.T) {
 			require.NoError(t, resp.Body.Close())
 		})
 	}
+
+	// Verifies s3 requests are signed without URL escaping to match AWS SDKs.
+	t.Run("disable URI escape when signing s3", func(t *testing.T) {
+		lp := createAWSAccessProxySuite(t, firstAWSCred)
+
+		// Avoid loading extra things.
+		t.Setenv("AWS_SDK_LOAD_CONFIG", "false")
+
+		// Create a real AWS SDK s3 client.
+		awsConfig := aws.NewConfig().
+			WithDisableSSL(true).
+			WithRegion(awsRegion).
+			WithCredentials(firstAWSCred).
+			WithEndpoint(lp.GetAddr()).
+			WithS3ForcePathStyle(true)
+
+		session, err := session.NewSession(awsConfig)
+		require.NoError(t, err)
+
+		s3client := s3.New(session)
+
+		// Use a bucket name with special charaters. AWS SDK actually signs the
+		// request with the unescaped bucket name.
+		_, err = s3client.ListObjects(&s3.ListObjectsInput{
+			Bucket: aws.String("=bucket=name="),
+		})
+		require.NoError(t, err)
```

Comment:
nit: It looks like a separate  independed  test flow. Can we move it to a separate Test ?